### ISSUE TITLE
XNNPACK COS returns values outside [-1, 1] for large float32 inputs

### ISSUE THREAD

**Issue by laolvfan** [google-ai-edge/LiteRT#9020]
## Environment

- LiteRT: `ai-edge-litert 2.1.6`
- TensorFlow converter: `tf-nightly 2.22.0-dev20260726`
- OS: Linux x86_64, CPU
- Python: 3.11

## Minimal reproduction

```python
import tensorflow as tf
from ai_edge_litert.interpreter import Interpreter, OpResolverType


class Model(tf.keras.Model):
    def call(self, x):
        return tf.math.cos(x)


x = tf.constant([1.0e10], dtype=tf.float32)
model = Model()
expected = model(x).numpy()
flatbuffer = tf.lite.TFLiteConverter.from_keras_model(model).convert()


def run(**kwargs):
    interpreter = Interpreter(model_content=flatbuffer, **kwargs)
    interpreter.allocate_tensors()
    input_detail = interpreter.get_input_details()[0]
    interpreter.set_tensor(input_detail["index"], x.numpy())
    interpreter.invoke()
    return interpreter.get_tensor(interpreter.get_output_details()[0]["index"])


print("eager:", expected)
print("LiteRT default:", run())
print(
    "LiteRT without default delegates:",
    run(experimental_op_resolver_type=OpResolverType.BUILTIN_WITHOUT_DEFAULT_DELEGATES),
)
```

## Actual result

```text
eager: [0.87311965]
LiteRT default: [-4901.761]
LiteRT without default delegates: [0.87311965]
```

The default LiteRT interpreter creates the XNNPACK CPU delegate. Its `COS`
path produces a value far outside the mathematical range of cosine. The same
FlatBuffer is correct with LiteRT's builtin interpreter without default
delegates.

## Expected result

Both LiteRT configurations should return a value in `[-1, 1]` and agree with
the eager result within normal float32 accuracy.

## Additional context

This was migrated from tensorflow/tensorflow#124080 after the TensorFlow issue
tracker directed TFLite reports to LiteRT.

**Comment by shivvaam0001** [comment]
Hi @laolvfan,

Thanks for reporting this issue. I have successfully run the script on my end and confirm that i observe the same behavior:
```
1. Standard TensorFlow :  0.8731196522712708
2. LiteRT with Default XNNPACK:  -4901.76123046875
3. LiteRT without XNNPACK:  0.8731196522712708
```
Quick Question: Have you observed issue with other trigonometric operations, such as `tf.math.sin` or `tf.math.tan` ?

**Comment by laolvfan** [comment]
Hi @shivvaam0001,

I have only verified the issue for `tf.math.cos` so far. I have not yet observed or systematically tested the same large-input behavior for `tf.math.sin` or `tf.math.tan`. I can run the same default-XNNPACK versus no-delegate comparison for those operations next.

**Comment by laolvfan** [comment]
Hi @shivvaam0001,

I tested `tf.math.sin` and `tf.math.tan` with the same setup (input `1.0e10`, default XNNPACK vs `BUILTIN_WITHOUT_DEFAULT_DELEGATES`):

**`sin`** reproduces the same pattern as `cos`:
```
eager: [-0.48750603]
LiteRT default (XNNPACK): [-4891.3433]
LiteRT without default delegates: [-0.48750603]
```

**`tan`** could not be tested this way — `tf.Tan` is not a native TFLite op, so conversion fails with `ERROR_NEEDS_FLEX_OPS` regardless of delegate, before it ever reaches the XNNPACK comparison. It's out of scope for this issue.

So this looks like a shared XNNPACK trigonometric range-reduction problem affecting at least `cos` and `sin`, not something specific to `cos`.

**Comment by aravindmurali22** [comment]
Hi @fengwuyao , following up on this thread, are there any recent updates on this? Thanks!
